nat-lab/tests/test_batching.py

```diff
@@ -0,0 +1,278 @@
+# Packet captures and histograms are used to visually and in-code observe the batching in action.
+import asyncio
+import pytest
+import telio
+from contextlib import AsyncExitStack
+from helpers import setup_mesh_nodes, SetupParameters
+from scapy.all import PcapReader  # type: ignore
+from scapy.layers.inet import IP, UDP, ICMP  # type: ignore
+from telio import PathType, State, AdapterType
+from telio_features import (
+    Batching,
+    LinkDetection,
+    PersistentKeepalive,
+    TelioFeatures,
+    Direct,
+    Wireguard,
+)
+from typing import List
+from utils.batching import PacketFilter, capture_histograms, draw_histogram
+from utils.connection import DockerConnection
+from utils.connection_util import DOCKER_GW_MAP, ConnectionTag, container_id
+
+
+def longest_zero_streak(nums):
```

Comment:
This could be moved to the histogram (i.e. "batching") module

Reply:
I removed it entirely for now. The goal of this PR is to push something through the door and then augment with necessary metrics